OHHTTPStubs/Sources/OHHTTPStubsResponse.m

```diff
@@ -135,23 +144,47 @@ -(instancetype)initWithFileAtPath:(NSString*)filePath
                        statusCode:(int)statusCode
                           headers:(nullable NSDictionary*)httpHeaders
 {
+    NSURL *fileURL = filePath ? [NSURL fileURLWithPath:filePath] : nil;
+    self = [self initWithFileURL:fileURL
+                      statusCode:statusCode
+                         headers:httpHeaders];
+    return self;
+}
+
+-(instancetype)initWithFileURL:(NSURL *)fileURL
+                    statusCode:(int)statusCode
+                       headers:(nullable NSDictionary *)httpHeaders {
     NSInputStream* inputStream;
-    if (filePath)
+    if (fileURL)
     {
-        inputStream = [NSInputStream inputStreamWithFileAtPath:filePath];
+        // [NSURL -isFileURL] is only available on iOS 8+
+        if (![fileURL.scheme isEqualToString:NSURLFileScheme]) {
+            @throw [NSException exceptionWithName:NSInternalInconsistencyException
+                                           reason:[NSString stringWithFormat:@"Only file URLs may be passed to %@", NSStringFromSelector(_cmd)]
+                                         userInfo:nil];
+        }
+        inputStream = [NSInputStream inputStreamWithURL:fileURL];
     }
     else
     {
-        NSLog(@"%s: nil file path. Returning empty data", __PRETTY_FUNCTION__);
+        NSLog(@"%s: nil URL. Returning empty data", __PRETTY_FUNCTION__);
         inputStream = [NSInputStream inputStreamWithData:[NSData data]];
     }
     
-    NSDictionary* attributes = [NSFileManager.defaultManager attributesOfItemAtPath:filePath error:nil];
-    unsigned long long fileSize = [[attributes valueForKey:NSFileSize] unsignedLongLongValue];
-    self = [self initWithInputStream:inputStream
-                            dataSize:fileSize
-                          statusCode:statusCode
-                             headers:httpHeaders];
+    NSError *error;
+    NSNumber *fileSize;
+    [fileURL getResourceValue:&fileSize forKey:NSURLFileSizeKey error:&error];
+    
+    if (error) {
+        NSLog(@"%s: Error getting file size for URL. Returning empty data", __PRETTY_FUNCTION__);
+        inputStream = [NSInputStream inputStreamWithData:[NSData data]];
+    } else {
+        self = [self initWithInputStream:inputStream
+                                dataSize:[fileSize unsignedLongLongValue]
+                              statusCode:statusCode
+                                 headers:httpHeaders];
+    }
+    
```

Comment:
Well, I'm not so sure. I think for some bizarre reasons the method could also fail while setting a non-nil value to `fileSize`. I mean, sure that's very unlikely of course, and would be strange. But nothing from the doc prevent it from happening.

So if the method return a success boolean, you're always supposed to test it anyway. That's the official pattern for all those methods taking an `NSError**` pioneer and returning a `BOOL`, you're supposed to test the returned `BOOL`, if it's `NO` then use the `NSError` to display the message, and if it's `YES` proceed with the data. In those methods, you're not supposed to test if `NSError` is `nil` or not to see if there is an error, but to test the returned `BOOL`. So I'd say that's the same for `fileSize`.

That's probably just style indeed because I don't think the method would even fill the `fileSize` pointer but still return `NO`, but that's how the method is defined and how the methods using `NSError**` and byref parameters are supposed to be used.